cot/src/email.rs

```diff
@@ -204,60 +325,116 @@ impl EmailBackend for SmtpEmailBackend {
+            EmailError::ConfigurationError(format!(
+                "Failed to validate SMTP configuration,error: {e}"
+            ))
+        })?;
+        let mut transport_builder = match &self.config.mode {
+            SmtpTransportMode::Localhost => SmtpTransport::relay("localhost").map_err(|e| {
+                EmailError::ConnectionError(format!(
+                    "Failed to create SMTP localhost transport,error: {e}"
+                ))
+            })?,
+            SmtpTransportMode::Unencrypted(host) => SmtpTransport::builder_dangerous(host),
+            SmtpTransportMode::Relay(host) => SmtpTransport::relay(host).map_err(|e| {
+                EmailError::ConnectionError(format!(
+                    "Failed to create SMTP relay transport host:{host},error: {e}"
+                ))
+            })?,
+            SmtpTransportMode::StartTlsRelay(host) => {
+                SmtpTransport::starttls_relay(host).map_err(|e| {
+                    EmailError::ConnectionError(format!(
+                        "Failed to create SMTP tls_relay transport host:{host},error: {e}"
+                    ))
+                })?
+            }
+        };
+        // Set the timeout for the transport
+        transport_builder = transport_builder.timeout(self.config.timeout);
+
+        // Set the port if provided in the configuration
+        // The port is optional, so we check if it's Some before setting it
+        // If the port is None, the default port for the transport will be used
+        if self.config.port.is_some() {
+            transport_builder = transport_builder.port(self.config.port.unwrap());
         }
-        let _socket_addr = format!("{}:{}", self.config.host, self.config.port)
-            .to_socket_addrs()
-            .map_err(|e| EmailError::ConnectionError(e.to_string()))?
-            .next()
-            .ok_or_else(|| {
-                EmailError::ConnectionError("Could not resolve SMTP host".to_string())
-            })?;
-
-        let mut transport_builder = SmtpTransport::builder_dangerous(&self.config.host)
-            .port(self.config.port)
-            .timeout(Some(self.config.timeout));
+
+        // Create the credentials using the provided configuration
+        let credentials = Credentials::try_from(&self.config).map_err(|e| {
+            EmailError::ConfigurationError(format!("Failed to create SMTP credentials,error: {e}"))
+        })?;
 
         // Add authentication if credentials provided
-        if let (Some(username), Some(password)) = (&self.config.username, &self.config.password) {
-            let credentials = Credentials::new(username.clone(), password.clone());
-            transport_builder = transport_builder.credentials(credentials);
+        let transport = if self.config.username.is_some() && self.config.password.is_some() {
+            transport_builder.credentials(credentials).build()
+        } else {
+            transport_builder.build()
+        };
```

Comment:
You already did that check above when creating `Credentials`, no need to duplicate it I think
```suggestion
        let transport = transport_builder.credentials(credentials).build();
```

Reply:
This is still current